woocommerce/woocommerce-gutenberg-products-block
# Filter Products by Attribute: Fix the page reload which happens when clicking the filter button on PHP templates

This PR fixes the page reload which is triggered immediately when you uncheck an attribute filter when the "Filter button" option is enabled.


Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6283
Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6282

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. On a block theme, go to Edit Site >  Product Catalog template.
2. Add Active Filters and Filter Products by Attribute block to the template.
3. Choose an attribute to filter, and make sure the filter button is toggled.
4. Go to the shop page.
5. Select an attribute, a reload should not happen until you click the filter button.
6. Once the filter has been applied. Uncheck it, a reload should not happen until you click the filter button.
7. Try this again with the filter button untoggled, although these reloads should now happen as soon as you check the attribute you're filtering by
8. Check there are no regressions on All Products block.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1.
2.
3.



### Performance Impact



### Changelog

> Filter Products by Attribute: Fix the page reload which happens when clicking the filter button on Woo templates using the Classic Template block
